### PR TITLE
feat: add retry logic for code checkout

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @bbbmj 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/worker/scm/manager.go [hunk 1]
```diff
@@ -206,7 +210,18 @@ func CloneRepo(token string, codeSource *api.CodeSource, ref string, folder stri
...
 
-	return p.Clone(token, url, reference, destPath)
+	var logs string
+	for i := 0; i < maxRetry; i++ {
+		logs, err = p.Clone(token, url, reference, destPath)
+		if err == nil {
+			return logs, nil
+		}
+
+		logdog.Infof("checkout code as error %v, retry times: %v", err, i+1)
```
Comment (supereagle) [inline pkg/worker/scm/manager.go]: Starting i from 1 can avoid `i+1` here.

Comment (zhujian7) [reply pkg/worker/scm/manager.go]: have fixed

**Pull request merged**